Source/BugsnagClient.m

```diff
@@ -843,6 +873,68 @@ - (void)updateCrashDetectionSettings {
         [self.oomWatchdog disable];
     }
 }
+
+/**
+ * Configure event listeners for automatic breadcrumbs.
+ */
+- (void)updateAutomaticBreadcrumbDetectionSettings {
+
+    // No automatic breadcrumbs are enabled; remove notifications for them all and return.
+    if (![[self configuration] enabledBreadcrumbTypes]) {
+        NSArray *allEventNames = [[[[self automaticBreadcrumbStateEvents]
+            arrayByAddingObjectsFromArray:[self automaticBreadcrumbControlEvents]]
+            arrayByAddingObjectsFromArray:[self automaticBreadcrumbMenuItemEvents]]
+            arrayByAddingObjectsFromArray:[self automaticBreadcrumbTableItemEvents]];
+        
+        for (NSString *eventName in allEventNames) {
+            [[NSNotificationCenter defaultCenter] removeObserver:self
+                                                            name:eventName
+                                                          object:nil];
+        }
+        return;
+    }
+    
+    // Register for the enabled automatic breadcrumbs
+
+    // 'State' events
+    if ([[self configuration] shouldRecordBreadcrumbType:BSGBreadcrumbTypeState]) {
+        for (NSString *name in [self automaticBreadcrumbStateEvents]) {
+            [self crumbleNotification:name];
+        }
+    }
+    
+    // 'User' events
+    if ([[self configuration] shouldRecordBreadcrumbType:BSGBreadcrumbTypeUser]) {
+        
+        // UI/NSTableView events
+        for (NSString *name in [self automaticBreadcrumbTableItemEvents]) {
+            [[NSNotificationCenter defaultCenter]
+                addObserver:self
+                   selector:@selector(sendBreadcrumbForTableViewNotification:)
+                       name:name
+                     object:nil];
+        }
+        
+        // UITextField/NSControl events (text editing)
+        for (NSString *name in [self automaticBreadcrumbControlEvents]) {
+            [[NSNotificationCenter defaultCenter]
+                addObserver:self
+                   selector:@selector(sendBreadcrumbForControlNotification:)
+                       name:name
+                     object:nil];
+        }
+        
+        // NSMenu events (Mac only)
+        for (NSString *name in [self automaticBreadcrumbMenuItemEvents]) {
```

Comment:
sendBreadcrumbForMenuItemNotification sets State type breadcrumbs

In fact, this seems questionable - might be worth raising a separate ticket to look at whether this should be navigation.